SCClassLibrary/External/Ableton/LinkClock.sc

```diff
@@ -0,0 +1,77 @@
+// clock with Ableton Link synchronization
+
+LinkClock : TempoClock {
+
+	var <>tempoChanged;
+
+	*newFromTempoClock { arg clock;
+		^super.new(
+			clock.tempo,
+			clock.beats,
+			clock.seconds,
+			clock.queue.maxSize
+		).initFromTempoClock(clock)
+	}
+
+	initFromTempoClock { arg clock;
+		var oldQueue;
+		//stop TempoClock and save its queue
+		clock.stop;
+		oldQueue = clock.queue.copy;
+		this.setMeterAtBeat(clock.beatsPerBar, 0);
+
+		forBy(1, oldQueue.size-1, 3) {|i|
+			var task=oldQueue[i+1];
+			//reschedule task with this clock
+			this.schedAbs(oldQueue[i], task);
+		};
+
+		^this
+	}
+
+
+	numPeers {
+		_LinkClock_NumPeers
+		^this.primitiveFailed
+	}
+
+	//override primitives
+	beats_ { arg beats;
+		_LinkClock_SetBeats
+		^this.primitiveFailed
+	}
+
+	// PRIVATE
```

Comment:
you should group together all private methods; if `setTempoAtBeat`, `setTempoAtSec` and`setMeterAtBeat` are meant to be private their names should start with `pr` too.

Reply:
These need to match TempoClock, don't they?

Reply:
@jamshark70 True, but if they are not private then they need to be moved out of this section. IMO they're not since they're not hidden in the schelp file and also not prefixed with `pr`.